lighthouse-cli/test/fixtures/offline-ready.html

```diff
@@ -24,18 +24,32 @@ <h4>
 </h4>
 
 <script>
-  if ('serviceWorker' in navigator) {
-    navigator.serviceWorker.register('/offline-ready-sw.js').then(function(registration) {
+  function sleep(ms) {
+    return new Promise(resolve => setTimeout(resolve, ms))
+  }
+
+  async function registerServiceWorker(qs = '') {
+    if (!('serviceWorker' in navigator)) return
+
+    try {
+      const registration = await navigator.serviceWorker.register(`/offline-ready-sw.js${qs}`)
       console.log('service worker registration complete');
 
       registration.addEventListener('statechange',  e => {
         console.log('sw registration is now', e.target.state);
       });
-    }).catch(function(e) {
+    } catch (e) {
       console.error('service worker is not so cool.', e);
       throw e;
-    });
+    }
   }
+
+  if (window.location.search.includes('slow')) {
+    setTimeout(() => registerServiceWorker('?delay=5000&slow'), 500);
```

Comment:
would it make any difference to do this on "load", where the load event is deferred by a significant amount of time?

Reply:
This `setTimeout` is mostly useless I can actually remove it since it's complicating the understanding of what's going on. Switching to fire on the load event is a good idea 👍 

the load event is currently being delayed by an image farther down that takes 2s, the only way LH doesn't catch service worker AFAICT is if the page doesn't call `.register` quickly enough, and this test was meant to prove that fact.

Reply:
AFAIK this (registering SW on "load") is basically what pintrest.com (which LH says has no SW) does. not sure if the key difference is how long until "load" fires - or something else entirely.

Reply:
If that's all they're doing, then I'm not sure what's going on there. We always wait until load so unless their load event handlers take a very long time such that we don't get to `.register` in time, then it must be something else.

Reply:
I can take a look closer at pinterest if there's not already too many cooks in the kitchen :)

Reply:
There's at most 1 :) IDK if ward ever started on it.

Reply:
Haha, alright I'll poke around.